DEPLOY.md

```diff
@@ -0,0 +1,101 @@
+# Deployment
+
+This repository contains two services: the RAP Agent and the RAP
+Controller. Both are deployed automatically on merge to main.
+
+
+## Agent deployment
+
+This runs within the secure backends (one instance per backend) and is
+managed by the [backend-server](https://github.com/opensafely-core/backend-server/) repo.
+
+A timer checks for new builds of the Docker image and pulls them down,
+so there may be a short delay (ten minutes maybe?) between the CI
+workflows completing and the new image being deployed.
+
+
+## Controller deployment
+
+We run a single instance of the controller which all the agents connect
+to. This is deployed as a Dokku app and at the time of writing runs on
+`dokku4`. A CI workflow connects to the host and runs the dokku deploy
+process with the newly built image.
+
+Below are the steps used to configure the app initially (though the
+exact config should be regarded as illustrative and may well have
+changed by the time of reading).
+
+Assume we have a server correctly configured by the [sysadmin](https://github.com/bennettoxford/sysadmin)
+tooling. This will ensure that Dokku is installed and configured with the
+necessary plugins and that a `rap-controller` user has been created.
+
+First we create the app and do some basic configuration:
+```bash
+dokku apps:create rap-controller
+
+# Set up the domain
+dokku domains:set rap-controller controller.opensafely.org
+dokku letsencrypt:enable rap-controller
+
+# Disable zero-dowtime deploys for service, we don't ever want two of
+# these loops running simultaneously
+dokku checks:disable rap-controller service
+
+# Create an appropriately owned storage directory and mount it
+sudo mkdir -p /var/lib/dokku/data/storage/rap-controller
+sudo chown rap-controller:rap-controller /var/lib/dokku/data/storage/rap-controller
+dokku storage:mount rap-controller /var/lib/dokku/data/storage/rap-controller:/storage
+
+# Run app as correct user, both the deployed services and one-off commands
+dokku docker-options:add rap-controller deploy,run "--user=$(id -u rap-controller):$(id -g rap-controller)"
```

Comment:
nit: this is fine & good. /ideally/ we would also add this user to the docker image, but afaik for us the only current downside is aesthetic:

```
madwort@dokku4:~$ dokku enter rap-controller web
groups: cannot find name for group ID 10002
I have no name!@e4c0461e1d26:/app$ whoami
whoami: cannot find name for user ID 10002
```
(I think we have tidied this up in some other containers but not all)

Reply:
Ah yeah, was discussing this earlier with Becky. IMNSHO adding users to the Docker image is the wrong way of dealing with this. Made an issue here to look at later:
https://github.com/opensafely-core/base-docker/issues/39

Reply:
ok let's discuss in that ticket